Clinical-Genomics/scout
# Redirect to login page when user session expires and user requests a report

fix #2008.

The error is caused by the fact that the session expires and the current_user is logged out. If the unlogged user is requesting one of the reports then the anonymous user object does not have the "admin" attribute any more. 

I'm fixing it in 2 ways:
- Simply redirect to login for all pages that are not static ("report" pages)
- Introduce an extra check in the method institute_and_case(that is used EVERYWHERE) to return HTTP error 401 page (unauthenticated) whenever the user is anonymous. (Might be an overkill but better safe than sorry)

**How to test**:
1. On a local instance of Scout, master branch, login user Clark Kent 
1. Go to a test case page and click on the case general report (HTML report is fine)
1. Go to the browser settings and remove the session cookies (or cookie)
1. Reload the report page and see that the page crashes.
1. Checkout to this branch and repeat the process.
1. The report page will redirect to the login page instead

**Expected outcome**:
Take a screenshot and attach or copy/paste the output.

**Review:**
